jenkins-master/jobs/tools/workspace/trigger.py

```
@@ -0,0 +1,143 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime
+import optparse
+import os
+import subprocess
+import sys
+import urllib
+import urllib2
+import zipfile
+
+CHUNK_SIZE = 8192
+
+
+class RunTPS():
+    def __init__(self, build_url, tests_url, debug=False):
+        self.build_url = build_url
+        self.debug = debug
+        self.tests_url = tests_url
+
+        # We declare the credentials here and we create the account
+        # after we install the fxa-python-client
+        self.username = 'fxa-%s@restmail.net' % os.urandom(6).encode('hex')
+        self.password = os.urandom(2).encode('hex')
+
+    def cleanup(self):
+        # Remove the account
+        subprocess.check_call(['fxa-client', '-e', self.username,
+                               '-p', self.password, 'destroy'])
+
+    def download(self, url, destination=''):
+        # Ensure we have a properly encoded URL
+        url = urllib.quote(url, safe='%/:=&?~#+!$,;\'@()*[]')
+
+        destination = os.path.abspath(destination)
+        try:
+            os.makedirs(destination)
+        except:
+            pass
+        file_name = os.path.join(destination, url.split('/')[-1])
```

You are hiding the real error here. This makes it very hard to investigate. In general plain except should be avoided.